src/lex/mod.rs

```diff
@@ -296,12 +293,10 @@ impl Lexer {
             return float.map(float_literal);
         }
         let literal = if self.match_next(b'u') || self.match_next(b'U') {
-            let unsigned = u64::try_from(digits)
-                .map_err(|_| "overflow while parsing unsigned integer literal")?;
+            let unsigned = u64::try_from(digits).map_err(|_| LexError::IntegerOverflow)?;
```

Comment:
Can you keep the information about whether it was signed or unsigned?

Reply:
The problem is that this same variant is also used in a place where the sign is not known. I could make it an `Option<bool>` if you like.

Reply:
I think the `Option<bool>` would be a good idea, yeah.